main.go

```diff
@@ -105,7 +105,11 @@ func main() {
 	if err := cmd.Wait(); err != nil {
 		if exitError := new(exec.ExitError); errors.As(err, &exitError) {
 			exitCode := exitError.ExitCode()
-			logErrorAndExit(exitCode, "Rspec exited with error %d", err)
+			if cfg.MaxRetries == 0 {
+				logErrorAndExit(exitCode, "Rspec exited with error %d", err)
+			} else {
+				retryFailedTests(testRunner, cfg)
```

Comment:
Do we want to add some logging here like 'Retrying, attempt X of Y' or something to show that a) retries are configured, and b) how many there are going to be?